### PR TITLE
[FluentInputBase] Use `Debouncer` instead of `PeriodicTimer` for debouncing `ValueChanged` handler with `ImmediateDelay`.

### PR DESCRIPTION
**Pull request by ApacheTech**
# Pull Request

## 📖 Description

This pull request applies the use of the internal [`Debouncer`](https://github.com/microsoft/fluentui-blazor/blob/dev/src/Core/Utilities/Debouncer.cs) instead of the `PeriodicTimer` to debounce values. A live demonstration has also been added to the `FluentSearch` page to show how `Immediate` can be used, as opposed to the `oninput` method.

### 🎫 Issues

#2030: Using the `FluentSearch` input box, but likely any debounce with `ImmediateDelay`, you can reliably force it to throw an exception, by firing the debounce while the `PeriodicTimer` is resetting itself. The exception shows that the `CancellationToken` used to reset the timer is set to cancelled, as the timer tries to tick.

## 👩‍💻 Reviewer Notes

The only code-flow change I've had to make is to push the invocation of the `ValueChanged` handler back to the Dispatcher, within `FluentInputBase.cs`. This is a thread safety precaution, as the use of the Debouncer takes the invocation of UI state changes off-thread.

## 📑 Test Plan

Because this is an implementation swap, any unit tests that cover the debouncing process will still pass, as intended. No further coverage necessary. Systems tests added by way of a working live example.

### General

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://www.fast.design/docs/community/contributor-guide) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have added [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for my new component
- [x] I have modified an existing component
- [x] I have validated the [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for an existing component 

## ⏭ Next Steps

It would be nice to expose the [`Debouncer`](https://github.com/microsoft/fluentui-blazor/blob/dev/src/Core/Utilities/Debouncer.cs) as public, and promote its use as a simple and effective debouncer, for most situations.

### CODE REVIEW

**Comment by ApacheTech** [comment]
@microsoft-github-policy-service agree

File: examples/Demo/Shared/Pages/Search/Examples/SearchImmediate.razor [hunk 1]
```diff
@@ -0,0 +1,110 @@
+﻿@using System.Timers;
```
Comment (dvoituron) [inline examples/Demo/Shared/Pages/Search/Examples/SearchImmediate.razor:1]: Could you remove this `using`, not required in the code?

File: examples/Demo/Shared/Pages/Search/Examples/SearchImmediate.razor [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                       Label="Immediate Delay"
+                       Placeholder="Delay"
+                       Min="0"
+                       Max="2000"
+                       Step="100" />
+
+    @* Search Box *@
+    <FluentSearch @bind-Value="_searchValue"
+                  @bind-Value:after="OnSearch"
+                  Immediate
```
Comment (dvoituron) [inline examples/Demo/Shared/Pages/Search/Examples/SearchImmediate.razor:17]: For "clean code", could you use the `Immediate="true"` syntax?

**Review by dvoituron — CHANGES_REQUESTED** [review]
Great job. Many thanks.

Could you update two details before merging? :-)

File: examples/Demo/Shared/Pages/Search/Examples/SearchImmediate.razor (hunk 1 above)
Comment (ApacheTech) [reply examples/Demo/Shared/Pages/Search/Examples/SearchImmediate.razor]: Done.

File: examples/Demo/Shared/Pages/Search/Examples/SearchImmediate.razor (hunk 2 above)
Comment (ApacheTech) [reply examples/Demo/Shared/Pages/Search/Examples/SearchImmediate.razor]: Done.

**Review by vnbaaij — APPROVED** [review]

**Review by dvoituron — APPROVED** [review]

**Pull request merged**